docs/en/routing/history-managers.md

```diff
@@ -0,0 +1,89 @@
+# History Managers
+
+Dojo Routing comes with three history managers for managing an application's navigation state, `HashHistory`, `StateHistory` and `MemoryHistory`. By default the `HashHistory` is used, however, this can be overridden by passing a different `HistoryManager` when creating the `Router` or using `registerRouterInjector`.
+
+>src/main.ts
+```ts
+import Router from '@dojo/framework/routing/Router';
+import StateHistory from '@dojo/framework/routing/history/StateHistory';
+
+import routes from './routes';
+
+// creates a router using the default history manager, `HashHistory`
+const router = new Router(routes);
+
+// creates a router using the `StateHistory`
+const routerWithHistoryOverride = new Router(routes, { HistoryManager: StateHistory });
+```
+
+Or using the `registerRouterInjector` helper function:
+
+>src/main.ts
+```ts
+import Registry from '@dojo/framework/widget-core/Registry';
+import { registerRouterInjector } from '@dojo/framework/routing/RouterInjector';
+import StateHistory from '@dojo/framework/routing/history/StateHistory';
+
+import routes from './routes';
+
+const registry = new Registry();
+
+// creates and registers a router using the default history manager, `HashHistory`
+registerRouterInjector(routes, registry);
+
+// creates and registers a router using the `StateHistory`
+registerRouterInjector(routes, registry, { HistoryManager: StateHistory });
+```
+
+## HashHistory
+
+ `HashHistory` uses the fragment identifier to process route changes, for example `https://foo.com/#home` would process the `home` as the route path. As `HashHistory` is the default manager, you do not need to import module.
+
+ ```ts
+import { Router } from '@dojo/framework/routing/Router';
+
+const router = new Router(config);
+```
+
+## StateHistory
+
+`StateHistory` uses the browser's [history API](https://developer.mozilla.org/en-US/docs/Web/API/History), to manage application route changes.
+
+**Note:** The `StateHistory` manager will require server-side machinery to enable an application to support refreshing on a route.
```

Comment:
Should tell what the server-side machinery is. For example:

1. modify the request path to load the static resource （js,css etc）
2. modify the request path to let the dojo routh redirct to home (SPA)

Reply:
we can also mention here that dojo build app's --serve mode has this machinery